src/System.Reflection.Emit/tests/AssemblyBuilderTests.cs

```diff
@@ -119,5 +232,46 @@ public void SetCustomAttribute_CustomAttributeBuilder_NullAttributeBuilder_Throw
             AssemblyBuilder assembly = Helpers.DynamicAssembly();
             Assert.Throws<ArgumentNullException>("customBuilder", () => assembly.SetCustomAttribute(null));
         }
+
+        public static IEnumerable<object[]> Equals_TestData()
+        {
+            AssemblyBuilder assembly = Helpers.DynamicAssembly(name: "Name1");
+            yield return new object[] { assembly, assembly, true };
+            yield return new object[] { assembly, Helpers.DynamicAssembly("Name1"), false };
+            yield return new object[] { assembly, Helpers.DynamicAssembly("Name2"), false };
+            yield return new object[] { assembly, Helpers.DynamicAssembly("Name1", access: AssemblyBuilderAccess.RunAndCollect), false };
+
+            yield return new object[] { assembly, new object(), false };
+            yield return new object[] { assembly, null, false };
+        }
+
+        [Theory]
+        [MemberData(nameof(Equals_TestData))]
+        public void Equals(AssemblyBuilder assembly, object obj, bool expected)
+        {
+            Assert.Equal(expected, assembly.Equals(obj));
+            if (obj is AssemblyBuilder)
+            {
+                Assert.Equal(expected, assembly.GetHashCode().Equals(obj.GetHashCode()));
+            }
+        }
+
+        public static void VerifyAssemblyBuilder(AssemblyBuilder assembly, AssemblyName name, IEnumerable<CustomAttributeBuilder> attributes)
+        {
+            Assert.Equal(name.ToString(), assembly.FullName);
+            Assert.Equal(name.ToString(), assembly.GetName().ToString());
+            
+            Assert.True(assembly.IsDynamic);
+
+            Assert.Equal(attributes?.Count() ?? 0, assembly.CustomAttributes.Count());
+
+            Assert.Equal(1, assembly.Modules.Count());
+            Module module = assembly.Modules.First();
+            Assert.Equal("<In Memory Module>", module.Name);
```

Comment:
Where is this module name defined?  I'm wondering if this string is coming from the product source and might be localized such that this could fail in non-English builds.